VirtoCommerce.CustomerModule.Data/Services/MemberServiceBase.cs

```diff
@@ -120,7 +120,23 @@ public virtual void SaveChanges(Member[] members)
                 }
 
                 CommitChanges(repository);
+
+                //for indexing new records we evaluate if this is a new record added through UI.
+                //if that's the case we will only have one record sent in this method and it's ID will be null.
+                bool isNewMember = false;
+                if (members.Count() == 1 && members.First().Id == null)
+                {
+                    isNewMember = true;
+                }
+
                 pkMap.ResolvePrimaryKeys();
+
+                //Now that the ID of the new record is resolved we publish the event for getting the new record indexed by MemberIndexationObserver.
+                if (isNewMember)
+                {
+                    MemberEventventPublisher.Publish(new MemberChangingEvent(EntryState.Added, members.First()));
```

Comment:
Not sure if previous event publishing is still required. Does bulk insert of records pass through this method?